src/main.cpp

```diff
@@ -3415,6 +3430,10 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
                 REJECT_INVALID, "orchard-commitment-tree-full");
         };
 
+        for (const auto& out : tx.vout) {
+            transparentValueDelta += out.nValue;
+        }
```

Comment:
I know this was mentioned in our pairing but I can't remember the answer. Why aren't `vpub_in` of JoinSplits, or negative valueBalances, treated like outputs from the transparent pool here, and conversely for inputs? If `transparentValueDelta` is counting what I think it is, then transfers to/from the shielded pools will not be taken into account when the should be.

![Screenshot_20230118-004413_Adobe Acrobat](https://user-images.githubusercontent.com/643204/213053217-25de67d0-8ce5-4389-bb40-af06768ed0d3.jpg)

Reply:
`transparentValueDelta` is simply the sum of the transparent outputs of the transaction (value entering the transparent pool) minus the sum of all transparent inputs (value leaving the transparent pool). If this value is negative in a block, then the value is flowing into one of the shielded pools, if it is positive in a block then value is coming out of the shielded pools and into transparent outputs.

Reply:
Oh, I see. When we do a transparent->shielded transfer in a JoinSplit/Spend/Action it must have been paid for by a transparent input, and that removes value from the transparent chain value pool. Conversely for shielded->transparent transfers which end up going to transparent outputs. So what I said would be double-counting the chain value pool delta.